src/main/java/bio/terra/service/auth/iam/IamService.java

```diff
@@ -148,6 +150,36 @@ public Map<UUID, Set<IamRole>> listAuthorizedResources(
     return callProvider(() -> iamProvider.listAuthorizedResources(userReq, iamResourceType));
   }
 
+  /**
+   * Note: calling this method will trigger a call to SAM API. No cache backs it. If calling,
+   * consider whether new usage patterns necessitate refactoring to add a cache.
```

Comment:
This is an interesting thought!  I went down this road initially, but abandoned it.  It seemed like we'd need to store `List<IamAction>` in the `AuthorizedCacheValue` rather than the key to properly support look ups, and then perform collection operations on every cache hit rather than fetching a boolean.  Wasn't sure if that added cost outweighed the possible reduction to SAM calls.

I can try this out though, and we can see how it looks / chat in design discussion tomorrow.

Reply:
@pshapiro4broad -- we talked about this more in Jade Core in stand-up today (@nmalfroy, @snf2ye).  What I hadn't considered was how this change to the existing cache would impact situations where a user had an action newly granted.  They could get caught with an out-of-date actions list until it expires from our cache.  This plus concerns around additional complexity in testing have us preferring to fetch individual actions from SAM in most cases.

But while investigating this, I did make some improvements to the existing cache that I'd like to keep.  I will write up a separate ticket to encompass that work rather than including it in this PR.

Reply:
Separate ticket: https://broadworkbench.atlassian.net/browse/DR-2642

Reply:
> @pshapiro4broad -- we talked about this more in Jade Core in stand-up today (@nmalfroy, @snf2ye). What I hadn't considered was how this change to the existing cache would impact situations where a user had an action newly granted.

@okotsopoulos I think this is incorrect; there is already code in place in TDR to invalidate the cache when a policy is changed via the TDR API. Look in `IamService` and search for calls to `authorizedMap.clear()`.

Reply:
@pshapiro4broad you're right!

@nmalfroy does this change your feelings on rewriting the cache?

Reply:
At any rate… there has been high enough volume of caching discussion that I will handle any caching changes as their own PR, whatever they will be :)

Reply:
Yeah, I think cache invalidation isn't really valid given that it's only for the given pod.  The problem is that you get into this weird in between state where some pods might have one permission and others will have different ones.  I kinda think we should actually be caching less

Reply:
D'oh, we have talked in the past about cache eviction and multiple pods, that's coming back to me now.  Thanks for refreshing my memory.

Here is my separate PR for improvements to the existing cache: https://github.com/DataBiosphere/jade-data-repo/pull/1303

Reply:
My assumption was that the original code was correct, so it was safe to extend it, I guess that was wrong :(

Looking back in time, this caching was added to speed up DRS lookups. https://broadworkbench.atlassian.net/browse/DR-1352
https://github.com/DataBiosphere/jade-data-repo/pull/620

Maybe a more narrowly focused cache that's only used on the DRS lookup code path would be a better solution here, so that an invalid cache state would have a smaller area of impact.

Reply:
100%.  I think DRS lookups are their own beast.  I suspect we're going to have a bit of perf tuning to do there in the coming months